aiida/storage/psql_dos/migrator.py

```diff
@@ -148,47 +146,144 @@ def validate_storage(self) -> None:
             stmt = select(DbSetting.val).where(DbSetting.key == REPOSITORY_UUID_KEY)
-            repo_uuid = connection.execute(stmt).scalar_one_or_none()
-            if repo_uuid is None:
+            database_repository_uuid = connection.execute(stmt).scalar_one_or_none()
+            if database_repository_uuid is None:
                 raise exceptions.CorruptStorage('The database has no repository UUID set.')
-            if repo_uuid != container_id:
+            if database_repository_uuid != repository_uuid:
                 raise exceptions.CorruptStorage(
-                    f'The database has a repository UUID configured to {repo_uuid} '
-                    f'but the disk-objectstore\'s is {container_id}.'
+                    f'The database has a repository UUID configured to {database_repository_uuid} '
+                    f'but the disk-objectstore\'s is {repository_uuid}.'
                 )
 
-    def initialise(self) -> None:
-        """Generate the initial storage schema for this profile, from the ORM models."""
-        from aiida.storage.psql_dos.backend import CONTAINER_DEFAULTS, get_filepath_container
+    def get_container(self) -> Container:
+        """Return the disk-object store container.
+
+        :returns: The disk-object store container configured for the repository path of the current profile.
+        """
+        from .backend import get_filepath_container
+        return Container(get_filepath_container(self.profile))
+
+    def get_repository_uuid(self) -> str:
+        """Return the UUID of the repository.
+
+        :returns: The repository UUID.
+        :raises: :class:`~aiida.common.exceptions.UnreachableStorage` if the UUID cannot be retrieved, which probably
+            means that the repository is not initialised.
+        """
+        try:
+            return self.get_container().container_id
+        except Exception as exception:
+            raise exceptions.UnreachableStorage(
+                f'Could not access disk-objectstore {self.get_container()}: {exception}'
+            ) from exception
+
+    def initialise(self, reset: bool = False) -> None:
+        """Initialise the storage backend.
+
+        This is typically used once when a new storage backed is created. If this method returns without exceptions the
+        storage backend is ready for use. If the backend already seems initialised, this method is a no-op.
+
+        :param reset: If ``true``, destroy the backend if it already exists including all of its data before recreating
+            and initialising it. This is useful for example for test profiles that need to be reset before or after
+            tests having run.
+        """
+        if reset:
+            self.reset_repository()
+            self.reset_database()
+
+        if not self.is_initialised:
+            self.initialise_repository()
+            self.initialise_database()
+
+        self.migrate()
```

Comment:
Migration is still needed, even for a freshly initialised database, correct?

For newcomers to the codebase it might make sense to add a brief comment here on what "migration" stands for

Reply:
I have added a clarifying comment